test/Elastic.Apm.AspNetFullFramework.Tests/TestsBase.cs

```diff
@@ -0,0 +1,509 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using AspNetFullFrameworkSampleApp.Controllers;
+using Elastic.Apm.Api;
+using Elastic.Apm.Config;
+using Elastic.Apm.Helpers;
+using Elastic.Apm.Logging;
+using Elastic.Apm.Model;
+using Elastic.Apm.Tests.MockApmServer;
+using Elastic.Apm.Tests.TestHelpers;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Elastic.Apm.AspNetFullFramework.Tests
+{
+	public class TestsBase : IAsyncLifetime
+	{
+		private static readonly string TearDownPersistentDataReason;
+
+		private static readonly bool TearDownPersistentData =
+			EnvVarUtils.GetBoolValue("ELASTIC_APM_TESTS_FULL_FRAMEWORK_TEAR_DOWN_PERSISTENT_DATA", /* defaultValue: */ true,
+				out TearDownPersistentDataReason);
+
+
+		protected readonly AgentConfiguration AgentConfig = new AgentConfiguration();
+
+
+		protected readonly bool SampleAppShouldHaveAccessToPerfCounters;
+
+		private readonly Dictionary<string, string> _envVarsToSetForSampleAppPool;
+		private readonly IisAdministration _iisAdministration;
+
+		private readonly IApmLogger _logger;
+		private readonly MockApmServer _mockApmServer;
```

Comment:
It's really awesome to have this in process! 👍

Reply:
I hope we will use it in other (not necessarily .NET Full Framework related) non-production projects.